.travis.yml

```diff
@@ -1,14 +1,49 @@
 language: rust
-services: docker
-sudo: required
+cache: cargo
 
-matrix:
-  include:
-    - rust: 1.27.0
-    - rust: stable
-    - rust: nightly
+rust:
+  - 1.27.0
+  - stable
+  - nightly
+
+install:
+  - rustup component add rustfmt
+  - rustup component add clippy
+  - rustup target add thumbv7em-none-eabihf
+  - rustup target add wasm32-unknown-unknown
 
 script:
-  - cargo test --all ; ./test_aesni.sh
+  - cargo test --all --release
+  - cargo test --all --all-features --release
 
-cache: cargo
+matrix:
+  allow_failures:
+    - rust: nightly
+  include:
+    - name: "Rust: stable (thumbv7em-none-eabihf)"
+      rust: stable
+      script:
+        - cargo build --all --target thumbv7em-none-eabihf --release
+    - name: "Rust: stable (wasm32-unknown-unknown)"
+      rust: stable
+      script:
+        - cargo build --all --target wasm32-unknown-unknown --release
```

Comment:
I am not sure if we need to bother with only building for WASM. I think we either need a full test run using Node, or we can drop it.

Reply:
If it builds on `thumbv7em-none-eabihf` it will probably build on `wasm32-unknown-unknown`, so seems ok to me

Reply:
Dropped in d5c1169